code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -42,12 +43,18 @@
 	var/door_anim_angle = 136
 	var/door_hinge = -6.5
 	var/door_anim_time = 2.0 // set to 0 to make the door not animate at all
+
 /obj/structure/closet/Initialize(mapload)
 	if(mapload && !opened)		// if closed, any item at the crate's loc is put in the contents
 		addtimer(CALLBACK(src, PROC_REF(take_contents)), 0)
 	. = ..()
+	populate_contents_immediate()
 	update_icon()
-	PopulateContents()
+
+/// Used to immediately fill a closet on spawn.
+/// Use this if you are spawning any items which can be tracked inside the closet.
+/obj/structure/closet/proc/populate_contents_immediate()
+	return
```

Comment:
Should objective items use this? GPS, etc? Feels bad to just throw this on and then not actually use it anywhere

Reply:
Doesn't really matter since objectives aren't trackable. TG have objectives in here, but I didn't port that since our objective items can't be tracked.

Reply:
It does matter and this system is now used, along with the unit test from TG being ported, with some modifications.